### PR TITLE
feat(nat/dnat): add new private dnat resource

### PR DESCRIPTION
**Pull request by Lance52259**
**What this PR does / why we need it**:
New resource support: private DNAT rule.

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
1. add new private dnat resource
```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/nat' TESTARGS='-run=TestAccPrivateDnatRule_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/nat -v -run=TestAccPrivateDnatRule_ -timeout 360m -parallel 4
=== RUN   TestAccPrivateDnatRule_basic
=== PAUSE TestAccPrivateDnatRule_basic
=== RUN   TestAccPrivateDnatRule_elbBackend
=== PAUSE TestAccPrivateDnatRule_elbBackend
=== RUN   TestAccPrivateDnatRule_vipBackend
=== PAUSE TestAccPrivateDnatRule_vipBackend
=== RUN   TestAccPrivateDnatRule_customIpAddress
=== PAUSE TestAccPrivateDnatRule_customIpAddress
=== CONT  TestAccPrivateDnatRule_basic
=== CONT  TestAccPrivateDnatRule_vipBackend
=== CONT  TestAccPrivateDnatRule_customIpAddress
=== CONT  TestAccPrivateDnatRule_elbBackend
--- PASS: TestAccPrivateDnatRule_customIpAddress (87.42s)
--- PASS: TestAccPrivateDnatRule_vipBackend (87.80s)
--- PASS: TestAccPrivateDnatRule_basic (240.81s)
--- PASS: TestAccPrivateDnatRule_elbBackend (261.12s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/nat       261.235s
```

### CODE REVIEW

File: docs/resources/nat_private_dnat_rule.md [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+
+* `region` - (Optional, String, ForceNew) Specifies the region where the DNAT rule is located.  
+  If omitted, the provider-level region will be used. Changing this will create a new resource.
+
+* `gateway_id` - (Required, String, ForceNew) Specifies the private NAT gateway ID to which the DNAT rule belongs.  
+  Changing this will create a new resource.
+
+* `transit_ip_id` - (Required, String) Specifies the ID of the transit IP for private NAT.
+
+* `transit_service_port` - (Optional, Int) Specifies the port of the transit IP.
```
Comment (Jason-Zhang9309) [inline docs/resources/nat_private_dnat_rule.md:120]: add a description of the default, for example `defaults to 0, which indicates any`

File: docs/resources/nat_private_dnat_rule.md [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+
+* `protocol` - (Optional, String) Specifies the protocol type.  
+  The valid values are **tcp**, **udp** and **any**. Defaults to **any**.
+
+* `backend_interface_id` - (Optional, String) Specifies the network interface ID of the transit IP for private NAT.  
+  Exactly one of `backend_interface_id` and `backend_private_ip` must be set.
+
+* `backend_private_ip` - (Optional, String) Specifies the private IP address of the backend instance.
+
+* `internal_service_port` - (Optional, Int) Specifies the port of the backend instance.
```
Comment (Jason-Zhang9309) [inline docs/resources/nat_private_dnat_rule.md:132]: ditto

File: huaweicloud/services/nat/resource_huaweicloud_nat_private_dnat_rule.go [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+		Protocol:            d.Get("protocol").(string),
+		Description:         d.Get("description").(string),
+		NetworkInterfaceId:  d.Get("backend_interface_id").(string),
+		PrivateIpAddress:    d.Get("backend_private_ip").(string),
+		InternalServicePort: strconv.Itoa(d.Get("internal_service_port").(int)),
+		TransitIpId:         d.Get("transit_ip_id").(string),
+		TransitServicePort:  strconv.Itoa(d.Get("transit_service_port").(int)),
+	}
+
+	log.Printf("[DEBUG] The create options of the transit IP (Private NAT) is: %#v", opts)
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/nat/resource_huaweicloud_nat_private_dnat_rule.go]:
```suggestion
	log.Printf("[DEBUG] The create options of the DNAT rule (Private NAT) is: %#v", opts)
```

File: huaweicloud/services/nat/resource_huaweicloud_nat_private_dnat_rule.go [hunk 4]
```diff
@@ -0,0 +1,232 @@
...
+	resp, err := dnats.Create(client, opts)
+	if err != nil {
+		return diag.Errorf("error creating transit IP (Private NAT): %s", err)
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/nat/resource_huaweicloud_nat_private_dnat_rule.go]: ditto

File: huaweicloud/services/nat/resource_huaweicloud_nat_private_dnat_rule.go [hunk 5]
```diff
@@ -0,0 +1,232 @@
...
+				Type:        schema.TypeInt,
+				Optional:    true,
+				Description: "The port of the transit IP.",
+			},
+			"protocol": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				Description: "The protocol type.",
+				ValidateFunc: validation.StringInSlice([]string{
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/nat/resource_huaweicloud_nat_private_dnat_rule.go]: should we remove the validate func?

File: docs/resources/nat_private_dnat_rule.md (hunk 1 above)
Comment (Lance52259) [reply docs/resources/nat_private_dnat_rule.md:120]: done.

File: docs/resources/nat_private_dnat_rule.md (hunk 2 above)
Comment (Lance52259) [reply docs/resources/nat_private_dnat_rule.md:132]: done.

File: huaweicloud/services/nat/resource_huaweicloud_nat_private_dnat_rule.go (hunk 3 above)
Comment (Lance52259) [reply huaweicloud/services/nat/resource_huaweicloud_nat_private_dnat_rule.go]: done.

File: huaweicloud/services/nat/resource_huaweicloud_nat_private_dnat_rule.go (hunk 5 above)
Comment (Lance52259) [reply huaweicloud/services/nat/resource_huaweicloud_nat_private_dnat_rule.go]: all validate functions have been deleted.

File: huaweicloud/services/nat/resource_huaweicloud_nat_private_dnat_rule.go (hunk 4 above)
Comment (Lance52259) [reply huaweicloud/services/nat/resource_huaweicloud_nat_private_dnat_rule.go]: done.

**Pull request merged**